src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final String DEFAULT_COLOR = Color.WHITE.name();
+
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private Figure figure;
```

Comment:
The variable 'figure' should not be a class-level field since it is only used in the local context of the methods. Consider declaring it as a local variable inside the methods where it's used.